src/services/assistants/search.ts

```diff
@@ -69,34 +91,36 @@ export class PromptSearchService extends VeridaService {
             console.log(`Searching by timeframe: ${maxDatetime} ${sort}`)
             if (promptSearchResult.databases.indexOf(SearchType.EMAILS) !== -1) {
-                emails = await searchService.schemaByDateRange<SchemaEmail>(SearchType.EMAILS, maxDatetime, sort, MAX_DATERANGE_EMAILS*3)
-                const emailShortlist = new EmailShortlist(llm)
-                emails = await emailShortlist.shortlist(prompt, emails, MAX_DATERANGE_EMAILS)
+                emails = await searchService.schemaByDateRange<SchemaEmail>(SearchType.EMAILS, maxDatetime, sort, config.dataTypes.emails.limit*3)
+                if (emails.length > config.dataTypes.emails.limit) {
+                    const emailShortlist = new EmailShortlist(llm)
+                    emails = await emailShortlist.shortlist(prompt, emails, config.dataTypes.emails.limit)
+                }
             }
             if (promptSearchResult.databases.indexOf(SearchType.FILES) !== -1) {
-                files = await searchService.schemaByDateRange<SchemaFile>(SearchType.FILES, maxDatetime, sort, MAX_DATERANGE_FILES)
+                files = await searchService.schemaByDateRange<SchemaFile>(SearchType.FILES, maxDatetime, sort, config.dataTypes.files.limit)
             }
             if (promptSearchResult.databases.indexOf(SearchType.FAVORITES) !== -1) {
-                favourites = await searchService.schemaByDateRange<SchemaFavourite>(SearchType.FAVORITES, maxDatetime, sort, MAX_DATERANGE_FAVORITES)
+                favourites = await searchService.schemaByDateRange<SchemaFavourite>(SearchType.FAVORITES, maxDatetime, sort, config.dataTypes.favorites.limit)
             }
             if (promptSearchResult.databases.indexOf(SearchType.FOLLOWING) !== -1) {
-                following = await searchService.schemaByDateRange<SchemaFollowing>(SearchType.FOLLOWING, maxDatetime, sort, MAX_DATERANGE_FOLLOWING)
+                following = await searchService.schemaByDateRange<SchemaFollowing>(SearchType.FOLLOWING, maxDatetime, sort, config.dataTypes.following.limit)
             }
             if (promptSearchResult.databases.indexOf(SearchType.CHAT_MESSAGES) !== -1) {
-                chatMessages = <SchemaSocialChatMessage[]> await searchService.schemaByDateRange(SearchType.CHAT_MESSAGES, maxDatetime, sort, MAX_DATERANGE_CHAT_MESSAGES)
+                chatMessages = <SchemaSocialChatMessage[]> await searchService.schemaByDateRange(SearchType.CHAT_MESSAGES, maxDatetime, sort, config.dataTypes.chatMessages.limit)
             }
             if (promptSearchResult.databases.indexOf(SearchType.CALENDAR_EVENT) !== -1) {
-                calendarEvents = await searchService.schemaByDateRange<SchemaEvent>(SearchType.CALENDAR_EVENT, maxDatetime, sort, MAX_DATERANGE_CALENDAR_EVENT)
+                calendarEvents = await searchService.schemaByDateRange<SchemaEvent>(SearchType.CALENDAR_EVENT, maxDatetime, sort, config.dataTypes.calendarEvents.limit)
             }
         }
```

Comment:
Room for performance improvements above as all the asynchronous calls are currently run in sequence but could be done in parallel with `Promise.allSettled`.